File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/auth/domain/Role.java
package com.example.growingshopauth.auth.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Embedded
    private Policies policies = new Policies();

    public Role(String name, List<Policy> policies) {
        this.name = name;
        this.policies = new Policies(policies);
    }

    public void changePolicies(Policies target) {
        this.policies = target;
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/domain/Role.kt
package com.example.growingshopauth.auth.domain

import com.example.growingshopcommon.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity

@Entity
class Role(
    @Column(nullable = false, unique = true)
    val name: String,

    @Embedded
    var policies: Policies = Policies()
) : BaseEntity() {
    fun changePolicies(target: Policies) {
        this.policies = target
    }
}
